### PR TITLE
[Storage] Make the executor interface take generic StateKey, Value fo…

### PR DESCRIPTION
**Pull request by sitalkedia**
…r fine grained storage support

## Motivation

Make the executor interface agnstic of AccountAdress and AccountBlob and instead accept a generic state_key, state_value pair. This is needed for supporting tables with fine-grained storage. The change is relatively big touching many files because of extensive API changes but core changes are in `apply_chunk_output.rs`, `state_view.rs`.  It also gets rid of the `get_by_access_path` and unify the state_view API to use `get_state_value` API.

Please note that currently deletion is supported by setting the state value to None - this is a bit hacky as it doesn't actually delete the node from JMT. We will have a follow up diff to support deletion of nodes in JMT

## Test Plan

Added a test to test insert and delete a raw key from state store.

### CODE REVIEW

File: docker/build-common.sh [hunk 1]
```diff
@@ -35,6 +35,9 @@ if [ "$IMAGE_TARGETS" = "release" ] || [ "$IMAGE_TARGETS" = "all" ]; then
           -p aptos-writeset-generator \
           "$@"
 
+  # Build our core modules!
```
Comment (sitalkedia) [inline docker/build-common.sh:38]: These are accidentally added changes - working on reverting them.

File: docker/init/Dockerfile [hunk 2]
```diff
@@ -32,17 +32,10 @@ RUN mkdir -p /opt/aptos/bin
 COPY --from=builder /aptos/target/release/aptos-genesis-tool /usr/local/bin
 COPY --from=builder /aptos/target/release/aptos-operational-tool /usr/local/bin
 
-### Get DPN Move modules bytecodes for genesis ceremony
```
Comment (sitalkedia) [inline docker/init/Dockerfile:35]: Same as above

File: execution/executor/src/components/apply_chunk_output.rs [hunk 3]
```diff
@@ -316,58 +304,135 @@ pub fn ensure_no_retry(to_retry: Vec<Transaction>) -> Result<()> {
     Ok(())
 }
 
+fn process_write_set_for_access_path(
```
Comment (msmouse) [inline execution/executor/src/components/apply_chunk_output.rs:307]: nit: it's really "process_access_path_write_op", right?

File: execution/executor/src/components/apply_chunk_output.rs [hunk 4]
```diff
@@ -316,58 +304,135 @@ pub fn ensure_no_retry(to_retry: Vec<Transaction>) -> Result<()> {
...
+                bail!("Write set should be a subset of read set.")
+            }
+            TransactionPayload::WriteSet(_) => (),
+        },
+        Transaction::StateCheckpoint => {}
+    }
+    Ok(())
+}
+
+fn process_write_set_for_state_key(
```
Comment (msmouse) [inline execution/executor/src/components/apply_chunk_output.rs:353]: process_state_key_write_op

File: execution/executor/src/components/apply_chunk_output.rs [hunk 5]
```diff
@@ -316,58 +304,135 @@ pub fn ensure_no_retry(to_retry: Vec<Transaction>) -> Result<()> {
...
+    write_op: WriteOp,
+) -> Result<()> {
+    let address = access_path.address;
+    let path = access_path.path;
+    match account_to_state.entry(address) {
+        hash_map::Entry::Occupied(mut entry) => {
+            update_account_state(entry.get_mut(), path, write_op);
+        }
+        hash_map::Entry::Vacant(entry) => {
+            ensure_txn_valid_for_vacant_entry(transaction)?;
```
Comment (msmouse) [inline execution/executor/src/components/apply_chunk_output.rs:321]: nice refactor

File: types/src/state_store/state_value.rs [hunk 6]
```diff
@@ -31,29 +31,37 @@ use serde::{Deserialize, Serialize};
...
 
 impl StateValue {
-    fn new(bytes: Vec<u8>) -> Self {
+    fn new(maybe_bytes: Option<Vec<u8>>) -> Self {
         let mut hasher = StateValueHasher::default();
-        hasher.update(&bytes);
+        if let Some(bytes) = &maybe_bytes {
+            hasher.update(bytes);
+        } else {
+            hasher.update(&[]);
```
Comment (msmouse) [inline types/src/state_store/state_value.rs:44]: For None we can just use `HashValue::zero()`? Or better, `StateValue` can be `StateValue(Option<StateValueInner>)`, and later when we remove the hack, `StateValueInner` becomes `StateValue`

File: execution/executor/src/components/apply_chunk_output.rs [hunk 7]
```diff
@@ -167,27 +167,15 @@ impl ApplyChunkOutput {
...
         } = state_cache;
 
         // Apply write sets to account states in the AccountCache, resulting in new account states.
-        let account_states = to_keep
+        let state_store_updates = to_keep
             .iter()
-            .map(|(t, o)| process_write_set(t, &mut accounts, o.write_set().clone()))
-            .collect::<Result<Vec<_>>>()?;
-        let state_store_updates = account_states
-            .par_iter()
```
Comment (msmouse) [inline execution/executor/src/components/apply_chunk_output.rs:179]: Nice that we'll be able to remove this costly serialization when we go fine grained.

**Review by msmouse — APPROVED** [review]

File: execution/executor/src/tests/mod.rs [hunk 8]
```diff
@@ -323,6 +331,160 @@ fn test_noop_block_after_reconfiguration() {
...
+    )])
+    .freeze()
+    .unwrap();
+
+    apply_transaction_by_writeset(
+        db,
+        vec![(transaction1, write_set1), (transaction2, write_set2)],
+    );
+
+    let state_value_from_db1 = db
```
Comment (msmouse) [inline execution/executor/src/tests/mod.rs:433]: state_value1_from_db

**Review by msmouse — APPROVED** [review]

**Comment by davidiw** [comment]
how on earth did I come to author this commit? :O

**Comment by sitalkedia** [comment]
@davidiw - I messed up the rebase for this PR, which was on top of your PR - of course I fixed the rebase issue, but still somehow github thinks that you are the author of the PR. This means you have to take the blame for anything this PR breaks :D :D

**Pull request closed without merging**